src/diff/index.js

```diff
@@ -41,15 +45,18 @@ export function diff(dom, parentDom, newVNode, oldVNode, context, isSvg, excessD
 
 	try {
 		outer: if (oldVNode.type===Fragment || newType===Fragment) {
-			diffChildren(parentDom, newVNode, oldVNode, context, isSvg, excessDomChildren, mounts, c);
+			diffChildren(parentDom, newVNode, oldVNode, context, isSvg, excessDomChildren, mounts, c, childDom);
 
 			// Mark dom as empty in case `_children` is any empty array. If it isn't
 			// we'll set `dom` to the correct value just a few lines later.
 			dom = null;
 
 			if (newVNode._children.length) {
 				dom = newVNode._children[0]._dom;
-				newVNode._lastDomChild = newVNode._children[newVNode._children.length - 1]._dom;
+
+				// If lastChild is a Fragment, use _lastDomChild, else use _dom
+				let lastChild = newVNode._children[newVNode._children.length - 1];
+				newVNode._lastDomChild = lastChild._lastDomChild || lastChild._dom;
```

Comment:
I was initially not sure why we need to set `_lastDomChild` here. After a few minutes of playing around I discovered that it is necessary to set that property when `newVNode` represents a `Fragment` node. Maybe we can mention that in the comment?

Reply:
nvm, the `if`-statement around it makes this obvious. Should've payed more attention :sweat_smile: